### PR TITLE
Putting instance name, machine name, and stream id into ping's header…

### PR DESCRIPTION
**Pull request by tokaplan**
…s for QPS's convenience. Allows QPS not to read the content of every Ping request.

### CODE REVIEW

File: Src/PerformanceCollector/Shared/Implementation/QuickPulse/QuickPulseServiceClient.cs [hunk 1]
```diff
@@ -204,7 +205,7 @@ private void WriteSamples(IEnumerable<QuickPulseDataSample> samples, string inst
             this.serializerDataPointArray.WriteObject(stream, monitoringPoints.ToArray());
         }
 
-        private HttpWebResponse SendRequest(string httpVerb, string path, Action<Stream> onWriteBody)
+        private HttpWebResponse SendRequest(string httpVerb, string path, bool includeHeaders, Action<Stream> onWriteBody)
```
Comment (zakimaksyutov) [inline Src/PerformanceCollector/Shared/Implementation/QuickPulse/QuickPulseServiceClient.cs]: Do we need optimize and not send this info for posts? Don't think it is a huge overhead. I'd rather simplify the logic here and send them always.

File: Src/PerformanceCollector/Shared/Implementation/QuickPulse/QuickPulseConstants.cs [hunk 2]
```diff
@@ -14,5 +14,20 @@ internal class QuickPulseConstants
...
+
+        /// <summary>
+        /// Instance name header.
+        /// </summary>
+        internal const string XMsQpsInstanceNameHeaderName = "x-ms-qps-instance-name";
+
+        /// <summary>
+        /// Stream id header.
+        /// </summary>
+        internal const string XMsQpsStreamIdHeaderName = "x-ms-qps-stream-id";
```
Comment (SergeyKanzhelev) [inline Src/PerformanceCollector/Shared/Implementation/QuickPulse/QuickPulseConstants.cs]: what is stream id?

File: Src/PerformanceCollector/Shared/Implementation/QuickPulse/QuickPulseServiceClient.cs (hunk 1 above)
Comment (tokaplan) [inline Src/PerformanceCollector/Shared/Implementation/QuickPulse/QuickPulseServiceClient.cs]: Well we read Post's body anyway, so I figured it's not that complicated :)

File: Src/PerformanceCollector/Unit.Tests/QuickPulse/QuickPulseServiceClientTests.cs [hunk 3]
```diff
@@ -23,7 +23,7 @@ public sealed class QuickPulseServiceClientTests : IDisposable
 
         private readonly List<Tuple<DateTimeOffset, MonitoringDataPoint>> samples = new List<Tuple<DateTimeOffset, MonitoringDataPoint>>();
 
-        private readonly List<Tuple<DateTimeOffset, MonitoringDataPoint>> pings = new List<Tuple<DateTimeOffset, MonitoringDataPoint>>();
+        private readonly List<Tuple<PingHeaders, MonitoringDataPoint>> pings = new List<Tuple<PingHeaders, MonitoringDataPoint>>();
```
Comment (SergeyKanzhelev) [inline Src/PerformanceCollector/Unit.Tests/QuickPulse/QuickPulseServiceClientTests.cs]: Are those headers changes much? Could they be set once in constructor or initialization of sender?

File: Src/PerformanceCollector/Shared/Implementation/QuickPulse/QuickPulseConstants.cs (hunk 2 above)
Comment (tokaplan) [inline Src/PerformanceCollector/Shared/Implementation/QuickPulse/QuickPulseConstants.cs]: A GUID that uniquely identifies a QuickPulse client. Even when there are multiple application instances hosted on the same machine, stream id will be unique for each of them.

File: Src/PerformanceCollector/Unit.Tests/QuickPulse/QuickPulseServiceClientTests.cs (hunk 3 above)
Comment (tokaplan) [inline Src/PerformanceCollector/Unit.Tests/QuickPulse/QuickPulseServiceClientTests.cs]: This is unit test, convenience is king :)

**Pull request merged**